pkg/manifests/types.go

```diff
@@ -19,377 +19,398 @@ import (
-	// with aditional Alertmanager instances.
-	// default: nil
+	// Configures how the Prometheus component communicates
+	// with additional Alertmanager instances.
+	// The default value is `nil`.
 	AlertmanagerConfigs []AdditionalAlertmanagerConfig `json:"additionalAlertmanagerConfigs,omitempty"`
-	// EnforcedLabelLimit per-scrape limit on the number of labels accepted for a sample. If more than this
-	// number of labels are present post metric-relabeling, the entire scrape will be treated as
-	// failed. 0 means no limit.
-	// default: 0
+	// Specifies a per-scrape limit on the number of labels accepted for a sample. 
+	// If the number of labels exceeds this limit after metric relabeling, 
+	// the entire scrape is treated as failed. 
+	// The default value is `0`, which means that no limit is set.
 	EnforcedLabelLimit *uint64 `json:"enforcedLabelLimit,omitempty"`
-	// EnforcedLabelNameLengthLimit per-scrape limit on the length of labels name that will be accepted for
-	// a sample. If a label name is longer than this number post metric-relabeling, the entire scrape
-	// will be treated as failed. 0 means no limit.
-	// default: 0
+	// Specifies a per-scrape limit on the length of a label name for a sample. 
+	// If a label name exceeds this limit after metric relabeling, 
+	// the entire scrape is treated as failed. 
+	// The default value is `0`, which means that no limit is set.
 	EnforcedLabelNameLengthLimit *uint64 `json:"enforcedLabelNameLengthLimit,omitempty"`
-	// EnforcedLabelValueLengthLimit per-scrape limit on the length of labels value that will be accepted for
-	// a sample. If a label value is longer than this number post metric-relabeling, the entire scrape will
-	// be treated as failed. 0 means no limit.
-	// default: 0
+	// Specifies a per-scrape limit on the length of a label value for
+	// a sample. If a label value exceeds this limit after 
+	// metric relabeling, the entire scrape is treated as failed.
+	// The default value is `0`, which means that no limit is set.
 	EnforcedLabelValueLengthLimit *uint64 `json:"enforcedLabelValueLengthLimit,omitempty"`
-	// EnforcedSampleLimit defines a global limit on the number of scraped samples that will be accepted.
-	// This overrides any SampleLimit set per ServiceMonitor or/and PodMonitor. It is meant to be
-	// used by admins to enforce the SampleLimit to keep the overall number of samples/series under the
-	// desired limit. Note that if SampleLimit is lower that value will be taken instead.
-	// default: 0
+	// Specifies a global limit on the number of scraped samples that will be accepted.
+	// This setting overrides any `SampleLimit` value set per `ServiceMonitor` or `PodMonitor`. 
+	// Administrators can use this setting to enforce a `SampleLimit` value and 
+	// keep the overall number of samples or series under the set limit. 
+	// However, if the `SampleLimit` value is lower than this limit,
+	// the `SampleLimit` value is used instead.
```

Comment:
The current wording is a bit confusing IMHO.

```
	// This setting overrides any `SampleLimit` value set per `ServiceMonitor` or `PodMonitor`.
...
	// However, if the `SampleLimit` value is lower than this limit,
	// the `SampleLimit` value is used instead.
```

```suggestion
	// This setting overrides the `sampleLimit` value set in any user-defined `ServiceMonitor` or `PodMonitor` object if the value is greater than `enforcedSampleLimit`. 
	// Administrators can use this setting to keep the overall number of samples under control.
```

Reply:
fixed